src/Session.test.ts

```diff
@@ -520,6 +543,30 @@ describe("push", () => {
 
 			eventsource = new EventSource(url);
 		}));
+
+		it("Throw if session is not connected", () =>
+			new Promise<void>((done) => {
+				server.on("request", (req, res) => {
+					const session = new Session(req, res);
+
+					session.on("disconnected", () => {
+						expect(() => {
+							session.push('test')
+						}).toThrow()
+
+						done();
+					});
+
+					session.on("connected", () => {
+						expect(() => {
+							session.push('test')
+						}).not.toThrow()
+						res.emit('close')
```

Comment:
```suggestion
						res.end();
```

Just makes more sense semantically, here. We want to test pushing after the session _and_ the underlying connection has closed (which they always should be in tandem.)